### PR TITLE
[Regression] The attachments list is not displayed on XWiki 14.4.3 (error loading data) #131

### PR DESCRIPTION
**Pull request by oanalavinia**
* copy the attachmentsjson.vm and attachment_macros.vm code to this macro, since the (complete) code is not present on XWiki < 14.8; this page is used only if the 2 mentioned vms are not present / do not have the correct version;
* added the move button translation + some css, since it's missing in versions before 14.6
* before 14.4, we also didn't had access to a liveData object from JS, in order to refresh the entries when needed, so the complete page is refreshed on those cases
* fix bug for upload fields, which weren't working after coming from the edit action

In short, this PR will make the attachments livedata to work on versions before 14.8. The only limitation is the upload and deleted actions will trigger a page reload, instead of a livedata reload (this only before 14.4)

### CODE REVIEW

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml [hunk 1]
```diff
@@ -176,12 +176,24 @@ require(['jquery', 'xwiki-l10n!xwiki-confluence-attachments-messages'], function
...
     if ($(e.target).prop('id').startsWith('confluenceAttachments')) {
       // Select the livedata above the upload form.
-      const associatedLivedata = $(e.target).closest('form').prevAll('.liveData')[0];
-      const liveDataObj = $(associatedLivedata).data('liveData');
-      liveDataObj.updateEntries();
+      const uploadForm = $(e.target).closest('form');
+      let associatedLivedata = uploadForm.prevAll('.liveData').first();
+      // This should be deleted after upgrading the XWiki parent to a version &gt;= 14.4.
+      associatedLivedata = associatedLivedata.length &gt; 0 ?
+        associatedLivedata : uploadForm.prevAll('.xwiki-livedata').first();
```
Comment (mflorea) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml:190]:
```suggestion
      let associatedLivedata = uploadForm.prevAll('.liveData, .xwiki-livedata').first();
```

You can look for both, which ever is found / first.

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml [hunk 2]
```diff
@@ -176,12 +176,24 @@ require(['jquery', 'xwiki-l10n!xwiki-confluence-attachments-messages'], function
     enhanceUploadInputs();
   })
 
+  $(document).on('xwiki:dom:updated', function(e, data) {
+    enhanceUploadInputs();
```
Comment (mflorea) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml:180]: Shouldn't you enhance only the part of the DOM that was updated? Or at least add a comment explaining why we enhance all inputs, including those that haven't been updated (and which may already be enhanced).

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml [hunk 3]
```diff
@@ -196,7 +208,10 @@ require(['jquery', 'xwiki-l10n!xwiki-confluence-attachments-messages'], function
    */
   $(document).on('click', '.confluenceAttachmentsMacro .attachmentActions .actiondelete', function(e) {
     e.preventDefault();
-    var modal = $(e.currentTarget).closest('.liveData').nextAll('.deleteConfluenceAttachment');
+    let liveData = $(e.currentTarget).closest('.liveData');
+    // This should be deleted after upgrading the XWiki parent to a version &gt;= 14.4.
+    liveData = liveData.length &gt; 0 ? liveData : $(e.currentTarget).closest('.xwiki-livedata');
```
Comment (mflorea) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml:213]:
```suggestion
    // The 'xwiki-livedata' CSS class is present only before XWiki version 14.4.
    let liveData = $(e.currentTarget).closest('.liveData, .xwiki-livedata');
```

File: xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml [hunk 4]
```diff
@@ -208,7 +223,9 @@ require(['jquery', 'xwiki-l10n!xwiki-confluence-attachments-messages'], function
     e.preventDefault();
     var modal = $(e.currentTarget).closest('.deleteConfluenceAttachment');
     var button = $(modal.data('relatedTarget'));
-    var liveData = button.closest('.liveData').data('liveData');
+    let liveData = button.closest('.liveData');
+    // This should be deleted after upgrading the XWiki parent to a version &gt;= 14.4.
+    liveData = liveData.length &gt; 0 ? liveData : button.closest('.xwiki-livedata');
```
Comment (mflorea) [inline xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/Attachments.xml:228]: Same as above.
